src/windows.rs

```diff
@@ -165,22 +108,33 @@ pub fn list() -> Result<Vec<TrashItem>, Error> {
             description: "`EnumObjects` set its output to None.".into(),
         })?;
         let mut item_vec = Vec::new();
-        let mut item_uninit = MaybeUninit::<*mut ITEMIDLIST>::uninit();
-        while peidl.Next(1, item_uninit.as_mut_ptr(), std::ptr::null_mut()) == S_OK {
-            let item = item_uninit.assume_init();
-            defer! {{ CoTaskMemFree(item as *mut c_void); }}
-            let id = get_display_name((&recycle_bin).into(), item, _SHGDNF::SHGDN_FORPARSING)?;
-            let name = get_display_name((&recycle_bin).into(), item, _SHGDNF::SHGDN_INFOLDER)?;
-
-            let orig_loc = get_detail(&recycle_bin, item, &SCID_ORIGINAL_LOCATION as *const _)?;
-            let date_deleted = get_date_unix(&recycle_bin, item, &SCID_DATE_DELETED as *const _)?;
-
-            item_vec.push(TrashItem {
-                id,
-                name: name.into_string().map_err(|original| Error::ConvertOsString { original })?,
-                original_parent: PathBuf::from(orig_loc),
-                time_deleted: date_deleted,
-            });
+
+        // TODO: take care of freeing this?
+        let mut item_uninit = MaybeUninit::<ITEMIDLIST>::uninit();
```

Comment:
I changed from `MaybeUninit::<*mut ITEMIDLIST>` to `MaybeUninit::<ITEMIDLIST>` to get things to compile and that was probably a bad idea. I need to spend some more time thinking about this.

Reply:
From what I remember this must be a slice of sorts now, but I wasn't able to produce a pointer to it. It's probably an unsafe-Rust question, but if we should all be stuck I am sure we will get help over at the `windows` repository.